### PR TITLE
#7061 Rename Page Editor "Remove" action to "Deactivate/Delete"

### LINKED ISSUE #7061: Change mod "Remove" to "Deactivate" in Page Editor
Context
---
- https://www.notion.so/pixiebrix/Simplify-Mod-Management-Actions-in-Page-Editor-6dc0345e953443e6b4764c20a124afd6?pvs=4#317a4b44ab00426289c2a8cb3c5c4529
- Some Page Editor creators have reported confusion with the "Remove" button on mod, thinking that they'll delete the mod if they click Remove

Definition of Done
---
- The "Remove" option should be replaced with "Deactivate" with an "X" icon instead of trashcan to match the language/iconography on the mods screen. This applies to:
  - Mod Packages
  - Standalone Mods
- The "Remove" action for mod components should remain the same (with a trashcan icon) because they are deleting the mod component.

### PR DESCRIPTION
**Pull request by mnholtz**
## What does this PR do?

- Closes #7061
- Changes copy of actions on mod action menu in the Page Editor
    - For Mods & Standalone Mods: "Remove" -> "Deactivate"
    - For Mod Components within Mods: "Remove" -> "Delete"
    - "Remove from Mod" -> "Move from Mod"
- Changes copy on "Delete", "Deactivate", and "Move from Mod" modals (see demo)
- Refactoring by renaming relevant components/hooks/vars, to include appropriate/consistent terminology like "Mod", "Deactivate", and "Delete"

## Reviewer Tips

- Primary functional changes include:
    - Addition of `onDeactivate` action on `ActionMenu`
    - Renaming of `onRemove` action to `onDelete` in `ActionMenu`
    - Renaming and splitting up the `useRemoveExtension` hook -> (`useDeactivateModComponent`, `useDeleteModComponent`)

## Demo

<div>
    <a href="https://www.loom.com/share/ad5e445acee54caebd930ad7c26568b7">
      <p>whales - Google Search - 8 December 2023 - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/ad5e445acee54caebd930ad7c26568b7">
      
    </a>
  </div>

## Future Work

- See slack discussion here about ideas for further improvements: https://pixiebrix.slack.com/archives/C05VCFGSLAY/p1701992308750199

## Checklist

- [x] Add tests
- [x] Designate a primary reviewer @grahamlangford

### CODE REVIEW

File: src/pageEditor/sidebar/modals/MoveFromModModal.tsx [hunk 1]
```diff
@@ -96,13 +99,17 @@ const RemoveFromRecipeModal: React.FC = () => {
...
         <Alert variant="warning">
           <FontAwesomeIcon icon={faExclamationTriangle} />
-          &nbsp;This will delete the starter brick. To restore it, use the reset
-          button.
+          &nbsp;The{" "}
+          <strong>
+            <FontAwesomeIcon icon={faHistory} fixedWidth /> Reset{" "}
+          </strong>
+          action located on the mod&apos;s three-dot menu can be used to restore
+          the starter brick <strong>before saving the mod</strong>.
```
Comment (mnholtz) [inline src/pageEditor/sidebar/modals/MoveFromModModal.tsx:112]: Screen Shot 2023-12-08 at 5 05 58 PM

File: src/pageEditor/hooks/useDeactivateMod.ts [hunk 2]
```diff
@@ -17,39 +17,39 @@
...
+    async ({ modId, shouldShowConfirmation = true }) => {
       if (shouldShowConfirmation) {
         const confirmed = await showConfirmation({
-          title: "Remove Mod?",
+          title: "Deactivate Mod?",
           message:
-            "You can reactivate mods from the PixieBrix Extension Console",
-          submitCaption: "Remove",
+            "This action will deactivate the mod and remove it from the Page Editor. You can reactivate or delete mods from the PixieBrix Extension Console.",
+          submitCaption: "Deactivate",
```
Comment (mnholtz) [inline src/pageEditor/hooks/useDeactivateMod.ts:52]: Screen Shot 2023-12-08 at 5 08 26 PM

File: src/pageEditor/sidebar/DynamicModComponentListItem.tsx [hunk 3]
```diff
@@ -62,20 +65,23 @@ type DynamicModComponentListItemProps = {
...
 
-  const isButton = extension.type === "menuItem";
+  const isActive = activeElement?.uuid === modComponentFormState.uuid;
+  const modId = modComponentFormState.recipe?.id;
+  const isSiblingOfActiveListItem = activeElement?.recipe?.id
+    ? modId === activeElement?.recipe?.id
+    : false;
+  const isChildOfActiveListItem = modId === activeModId;
+  const isRelativeOfActiveListItem =
+    !isActive && (isChildOfActiveListItem || isSiblingOfActiveListItem);
```
Comment (mnholtz) [inline src/pageEditor/sidebar/DynamicModComponentListItem.tsx:81]: Functional logic should have remained the same here; refactoring to make this logic more self-explanatory than it was before.

File: src/pageEditor/hooks/useRemoveModComponent.ts [hunk 4]
```diff
@@ -82,13 +76,64 @@ function useRemoveExtension(): (useRemoveConfig: Config) => Promise<void> {
...
         });
       }
     },
-    [dispatch, sessionId, showConfirmation],
+    [dispatch, sessionId],
   );
 }
 
-export default useRemoveExtension;
+export const useDeactivateModComponent = (): ((
```
Comment (twschiller) [inline src/pageEditor/hooks/useRemoveModComponent.ts:88]: Should this be useDeactivateStandaloneMod?

File: src/pageEditor/hooks/useDeactivateMod.ts (hunk 2 above)
Comment (grahamlangford) [reply src/pageEditor/hooks/useDeactivateMod.ts:52]: Suggestion: Update `ConfirmationModal` to allow the `message` prop to be `string | ReactElement`, then make `useDeactivateMod` and `useRemoveModComponent` render a link to the Extension Console:
`message: (
            <>
              This action will deactivate the mod and remove it from the Page
              Editor. You can reactivate or delete mods from the{" "}
              <a
                href="#"
                onClick={async (e) => {
                  e.preventDefault();
                  await browser.runtime.openOptionsPage();
                }}
              >
                Extension Console
              </a>
              .
            </>
          ),`

File: src/pageEditor/hooks/useRemoveModComponent.ts (hunk 4 above)
Comment (grahamlangford) [reply src/pageEditor/hooks/useRemoveModComponent.ts:88]: Agreed. I would interpret `useRemoveModComponent` as the hook that's called when a ModComponent is removed from a packaged mod.

Comment (mnholtz) [reply src/pageEditor/hooks/useRemoveModComponent.ts:88]: The problem that it [sometimes _is_ used as an actual deactivate](https://github.com/pixiebrix/pixiebrix-extension/pull/7082/files#diff-df03ff8607f818b29d4c0f5a32e1c19fb369236fa6ccc646f64f6155c0e4db6aR67-R69). I originally had called it `useDeactivateStandaloneMod` until I had to change it's usage there.

Comment (grahamlangford) [reply src/pageEditor/hooks/useRemoveModComponent.ts:88]: That makes sense then. Yet another reason why I wish we didn't have the concept of StandaloneMod

File: src/components/ConfirmationModal.tsx [hunk 5]
```diff
@@ -27,7 +27,7 @@ import { type ButtonVariant } from "react-bootstrap/types";
 
 type ModalProps = {
   title?: string;
-  message: string;
+  message: string | React.ReactNode;
```
Comment (grahamlangford) [inline src/components/ConfirmationModal.tsx:30]:
```suggestion
  message: React.ReactNode;
```
`string` is valid for `ReactNode`. It also allows `boolean`, `number`, and several other types.

A more narrow type would be:
```suggestion
  message: string | React.ReactElement;
```

**Review by grahamlangford — APPROVED** [review]

File: src/components/ConfirmationModal.tsx (hunk 5 above)
Comment (mnholtz) [reply src/components/ConfirmationModal.tsx]: TIL the difference between the two, thank you for pointing this out

**Pull request merged**